### PR TITLE
chore: add Done method to streamers and renderer to exit successfully

### PR DESCRIPTION
**Pull request by efekarakus**
Previously, streamers and components from the `stream` and `progress`
packages respectively only exited if a `ctx` was canceled.

This PR enables `StackStreamer` to exit successfully on its own without
waiting for a context to cancel it when the stack is no longer in
progress.

Similarly, components that can both `Render` and `Listen` on events now
have a `Done` method to indicate that there are no more events to
`Listen` and we can exit successfully.

_By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice._

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/cloudformation.go [hunk 1]
```diff
@@ -23,6 +24,11 @@ import (
 	"golang.org/x/sync/errgroup"
 )
 
+const (
+	// waitForStackTimeout is how long we're willing to wait for a stack to go from in progress to a complete state.
+	waitForStackTimeout = 1*time.Hour + 30*time.Minute
```
Comment (iamhopaul123) [inline internal/pkg/deploy/cloudformation/cloudformation.go:29]: Wouldn't the timeout be too long 🤔

**Review by huanjani — APPROVED** [review]
LGTM!

**Review by iamhopaul123 — APPROVED** [review]
LGTM!

File: internal/pkg/deploy/cloudformation/cloudformation.go (hunk 1 above)
Comment (efekarakus) [reply internal/pkg/deploy/cloudformation/cloudformation.go:29]: Creating databases can take forever >_>

**Pull request merged**